### DISCUSSION TITLE
doesObjectExist() API

### DISCUSSION THREAD

**Discussion by sweta96** [Robothy/local-s3#34] [Q&A] [answered]
The true scenario for this API works, that is, if object exists, the method returns true.

Whereas, if object does not exists, it does not work.

Could you check, and let me know, if the implementation does not exists.

**Comment by Robothy** [comment]
LocalS3 supports this API. Please provide sample code if you thought it doesn't work.

**Comment by sweta96** [comment]
client.putObject(...)
client.deleteObject(...)
client.doesObjectExist(...)

Error: Method Not Allowed

**Comment by Robothy** [comment, marked as answer]
You found a bug. It'll be fixed in the next release.

A new test case is added.

https://github.com/Robothy/local-s3/blob/a2280031f2c45cbe2aa1022ada3350021cca86b6/local-s3-interationtest/src/test/java/com/robothy/s3/test/HeadObjectIntegrationTest.java#L11-L21
